test/SDiagonal.jl

```diff
@@ -70,6 +70,15 @@ using StaticArrays, Test, LinearAlgebra
 
         @test length(m) === 4*4
 
+        m2 = SMatrix{4,4}(m)
+        @test axes(m) === axes(m2)
```

Comment:
Does `@test (@inferred axes(m)) === axes(m2)` fail for some version of Julia?

Reply:
Seems to work, so I've included this now